src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembersAnalyzer.cs

```diff
@@ -153,6 +178,56 @@ static DiagnosticId GetDiagnosticId (SymbolValue source, SymbolValue target)
 				_ => throw new NotImplementedException ()
 			};
 
+		static Diagnostic ReportRequiresUnreferencedCodeDiagnostic (AttributeData requiresAttributeData, ISymbol member, Location location)
+		{
+			var message = RequiresUnreferencedCodeUtils.GetMessageFromAttribute (requiresAttributeData);
+			var url = RequiresAnalyzerBase.GetUrlFromAttribute (requiresAttributeData);
+			return Diagnostic.Create (
+				DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.RequiresUnreferencedCode),
+				location,
+				member.GetDisplayName (),
+				message,
+				url);
+		}
+
+		static IEnumerable<Diagnostic> VerifyMethodSymbolForDiagnostic (IMethodSymbol methodSymbol, Location location)
+		{
+			foreach (var parameter in methodSymbol.Parameters) {
+				if (parameter.GetDynamicallyAccessedMemberTypes () != DynamicallyAccessedMemberTypes.None)
+					yield return Diagnostic.Create (DiagnosticDescriptors.GetDiagnosticDescriptor (DiagnosticId.DynamicallyAccessedMembersMethodAccessedViaReflection), location, methodSymbol.GetDisplayName ());
```

Comment:
I think we should avoid yielding multiple diagnostics - there should be just one diagnostic for the method if any of its parameters or the return type on a virtual have annotations.

Reply:
I created a piece of code that I actually didnt like much which does yield return and then yield break to keep the IEnumerator and use common code. Maybe I should divide the Property that yields an IEnumerator from the method and field analysis which should return only one diagnostic. Let me know what you think